### PR TITLE
Contracts

### PR DESCRIPTION
**Pull request by makoto**

### CODE REVIEW

**Comment by makoto** [comment]
@xavierlepretre thanks for all the comments. Any specific reason you comment against each commit rather than the [PR files](https://github.com/makoto/secret_prediction/pull/1/files)? It's quite hard to follow all your comment.

**Comment by xavierlepretre** [comment]
Ah, it's because I cloned your repo, looked at the source in full, then clicked "Blame" to find the commit that had the line I wanted to commit on.
I had to look at the source in full to understand in the first place.
